lib/ws_servers/api/start_connections.js

```diff
@@ -31,42 +36,71 @@ class ConnectionManager {
     this.starting = true
 
     try {
-      const { ws, apiKey, apiSecret, dmsScope } = args
-      const newCredentials = apiKey !== this.apiKey || apiSecret !== this.apiSecret
+      const { ws, apiKey, apiSecret, dmsScope, mode } = args
+      const currentCredentials = this.credentials[mode]
+      const hasNewCredentials = apiKey !== currentCredentials.apiKey || apiSecret !== currentCredentials.apiSecret
       const hasClients = ws.algoWorker.isOpen && ws.clients.bitfinex && ws.clients.bitfinex.isOpen
 
-      this.apiKey = args.apiKey
-      this.apiSecret = args.apiSecret
-      this.scope = dmsScope
-
+      this._updateCredentials({ mode, apiKey, apiSecret, dmsScope })
       await this._startDmsControl(args)
 
-      if (newCredentials || !hasClients) {
-        await this._startAlgoWorker(args, newCredentials)
-        this._startBfxClient(args, newCredentials)
+      if (hasClients && !hasNewCredentials) {
```

Comment:
`hasClients` should be checked based on the mode as well; or not.. since `_startAlgoWorker` and `_startBfxClient` already does this .. so, may be we can just call these two functions with the ` hasNewCredentials`